src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -134,6 +134,21 @@ private void createTopic(final String topic, final int partitionsNum, final int
         } catch (final Exception e) {
             throw new TopicCreationException("Unable to create topic " + topic, e);
         }
+        // Next step is to wait for topic initialization. On can not skip this task, cause kafka instances may not
+        // receive information about topic creation, which in turn will block publishing.
+        // This kind of behavior was observed during tests, but may also present on highly loaded event types.
+        final long timeoutMillis = TimeUnit.SECONDS.toMillis(5);
+        final long borderTime = System.currentTimeMillis() + timeoutMillis;
+        boolean creationConfirmed = false;
+        while (!creationConfirmed && borderTime >= System.currentTimeMillis()) {
+            try (Consumer<byte[], byte[]> consumer = kafkaFactory.getConsumer()) {
+                final List<PartitionInfo> partitions = consumer.partitionsFor(topic);
+                creationConfirmed = null != partitions;
+            }
+        }
```

Comment:
1) I don't insist, the code you wrote is fine for me. However I think something like this would be more clear:
```java
final Boolean created = executeWithRetry(() -> {
            try (Consumer<byte[], byte[]> consumer = kafkaFactory.getConsumer()) {
                return consumer.partitionsFor(topic) != null;
            }
        },
        new RetryForSpecifiedTimeStrategy<Boolean>(TimeUnit.SECONDS.toMillis(5))
                .withResultsThatForceRetry(Boolean.FALSE));
if (created != Boolean.TRUE) {
    throw new TopicCreationException("Failed to confirm topic creation within " + timeoutMillis + " millis");
}
```

2) Actually I think that it would be better here to use interval here (and executeWithRetry allows that) to have a control over calling kafka method.

3) Why do you use consumer here not the producer? Creation of producer is not a fast thing.